custom_random_forest.py

```diff
@@ -0,0 +1,96 @@
+from concurrent.futures import ProcessPoolExecutor, ThreadPoolExecutor
+
+import numpy as np
+from sklearn.base import BaseEstimator
+from sklearn.tree import DecisionTreeClassifier
+
+
+class RandomForestClassifierCustom(BaseEstimator):
+    def __init__(self, n_estimators=10, max_depth=None, max_features=None, random_state=None):
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.max_features = max_features
+        self.random_state = random_state
+
+        self.trees = []
+        self.feat_ids_by_tree = []
+
+    def fit_each_tree(self, args) -> tuple:
```

Comment:
```suggestion
    def _fit_one_tree(self, args) -> tuple:
```
1) Мне кажется `one` тут лучше подходит чем `each`
2) Еще здорово тут добавить `_`, это ведь не часть публичного интерфейса